slice-core/src/main/java/com/cognifide/slice/core/internal/filter/ContextRequestFilter.java

```diff
@@ -124,4 +126,18 @@ public void init(final FilterConfig filterConfig) throws ServletException {
 	public void destroy() {
 	}
 
+    @Override
+    public void injectorCreated(Injector injector, InjectorConfig config) {
+        synchronized(this){

```

Comment:
Use `CopyOnWriteArrayList` for the `injectorNames` and remove `synchronized()` statements.